### PR TITLE
[ML] Fix job.services transformation for pipeline jobs

### PR DESCRIPTION
**Pull request by TonyJ1**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Comment by 0mza987** [comment]
after this [PR](https://github.com/Azure/azure-sdk-for-python/pull/27335) merged, you may need to merge the change and update the recording of target test

**Comment by TonyJ1** [comment]
> after this [PR](https://github.com/Azure/azure-sdk-for-python/pull/27335) merged, you may need to merge the change and update the recording of target test



> after this [PR](https://github.com/Azure/azure-sdk-for-python/pull/27335) merged, you may need to merge the change and update the recording of target test

I have merged with your changes and have enabled test_pipeline_job_create test and updated the recording.

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/pipeline/pipeline_job.py [hunk 1]
```diff
@@ -477,6 +477,7 @@ def _to_rest_object(self) -> JobBase:
             inputs=to_rest_dataset_literal_inputs(built_inputs, job_type=self.type),
             outputs=to_rest_data_outputs(built_outputs),
             settings=settings_dict,
+            services={k: v._to_rest_object() for k, v in self.services.items()} if self.services else None,
```
Comment (0mza987) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/pipeline/pipeline_job.py:480]: Please add a ut to test the lines added in this PR. Can add in test_pipeline_job_entity.py to test pipeline job to rest obj and rest obj to pipeline job

**Comment by 0mza987** [comment]
Seems there are some failed tests, please fix then locally and update the recordings to pass CI

**Review by wangchao1230 — APPROVED** [review]

File: sdk/ml/azure-ai-ml/tests/recordings/pipeline_job/e2etests/test_pipeline_job.pyTestPipelineJobtest_pipeline_job_create.json [hunk 2]
```diff
@@ -7,32 +7,32 @@
         "Accept": "application/json",
         "Accept-Encoding": "gzip, deflate",
         "Connection": "keep-alive",
-        "User-Agent": "azure-ai-ml/1.1.0 azsdk-python-mgmt-machinelearningservices/0.1.0 Python/3.8.13 (macOS-13.0-arm64-i386-64bit)"
+        "User-Agent": "azure-ai-ml/1.1.0 azsdk-python-mgmt-machinelearningservices/0.1.0 Python/3.7.8 (Windows-10-10.0.22621-SP0)"
```
Comment (wangchao1230) [inline sdk/ml/azure-ai-ml/tests/recordings/pipeline_job/e2etests/test_pipeline_job.pyTestPipelineJobtest_pipeline_job_create.json:10]: please resolve conflict detected in this file

Comment (0mza987) [reply sdk/ml/azure-ai-ml/tests/recordings/pipeline_job/e2etests/test_pipeline_job.pyTestPipelineJobtest_pipeline_job_create.json]: Resolved

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/pipeline/pipeline_job.py (hunk 1 above)
Comment (0mza987) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/pipeline/pipeline_job.py:485]: @TonyJ1 will add test in next PR

**Pull request merged**